extensions-core/s3-extensions/src/test/java/org/apache/druid/data/input/s3/S3InputSourceTest.java

```diff
@@ -535,6 +539,48 @@ public void testReader() throws IOException
     EasyMock.verify(S3_CLIENT);
   }
 
+  @Test(expected = SdkClientException.class)
+  public void testReaderRetriesOnSdkClientExceptionButNeverSucceedsThenThrows() throws Exception
```

Comment:
This tests the new custom resettable condition for S3Entity. However this test is fairly time consuming since it retries 10 times, the default. I found no easy way to avoid the 10 retries due to the structure of the code. Let me know if anyone can think of something else.

Reply:
Hm, the test took 6m when I tried it, which feels excessive. I think it'd be fine to modify `ObjectSummaryIterator` to have a fetch retry count parameter, which you could set from S3InputSource (maybe in a test-only path if you don't feel like exposing that in the JSON constructor)

Reply:
Done. It is kind of ugly though. In general I think we should avoid having static constants .